dali/test/python/test_external_source_cupy.py

```diff
@@ -55,3 +55,22 @@ def generator(i):
     with check_output_pattern(pattern):
         for _ in range(iterations):
             pipe.run()
+
+
+def _test_cross_device(src, dst):
+    import nvidia.dali.fn as fn
+    pipe = Pipeline(1, 3, dst)
+    with cp.cuda.Device(src):
+        data = cp.array([[1,2,3,4],[5,6,7,8]], dtype=cp.float32)
+    with pipe:
+        pipe.set_outputs(fn.external_source([data], batch=False, device='gpu', cycle=True))
+
+    pipe.build()
+    for _ in range(10):
+        pipe.run()
```

Comment:
Maybe you can validate the pipeline output?

Reply:
Well, I guess...